src/intrinsics/prepack/global.js

```diff
@@ -551,6 +551,58 @@ export default function(realm: Realm): void {
     })
   );
 
+  // Helper function that replaces the body of a source function with another source function body.
```

Comment:
You are not just replacing the body but the parameters too and much else beside.

Reply:
The body and everything that the body may depend on. Do you have a better name in mind?